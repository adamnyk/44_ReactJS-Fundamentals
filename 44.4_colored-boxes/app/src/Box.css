.Box {
	margin: 3px;
	height: 50px;
	width: 50px;
	display: flex;
	justify-content: center;
	align-items: center;
	color: white;
}
