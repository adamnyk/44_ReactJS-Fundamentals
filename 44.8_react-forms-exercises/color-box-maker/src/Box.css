.Box {
	display: flex;
	align-items: center;
	margin: 10px;
}

.Box-box {
	border-radius: 10px;
}

.Box-button {
	margin: 0 5px;
	background-color: darkgrey;
	border: none;
	border-radius: 3px;

}
