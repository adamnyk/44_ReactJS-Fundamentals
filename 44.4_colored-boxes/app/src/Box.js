import './Box.css'

const Box = (props) => {

    return (
        <div
            className="Box"
            style={{ backgroundColor: props.color }}
        >
            {props.msg}
        </div>
            
    )
}

export default Box