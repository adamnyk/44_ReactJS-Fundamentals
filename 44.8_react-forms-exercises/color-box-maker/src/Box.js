import "./Box.css";

const Box = ({
	color = "deepskyblue",
	height = 200,
	width = 200,
	id,
	remove
}) => {
	return (
		<div className="Box">
			<div
				className="Box-box"
				data-testid="colored-box"
				style={{
					backgroundColor: color,
					height: height + "px",
                    width: width + "px",
				}}
            >
            </div>
            <button
                className='Box-button'
				onClick={() => remove(id)}
            >
            x
            </button>
		</div>
	);
};

export default Box;
