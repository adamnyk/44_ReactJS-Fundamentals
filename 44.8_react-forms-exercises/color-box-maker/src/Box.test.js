import { render } from "@testing-library/react";
import Box from "./Box";

it("Box renders", () => {
	render(<Box />);
});

it("matches snapshot", function () {
	const { asFragment } = render(<Box />);
	expect(asFragment()).toMatchSnapshot();
});
