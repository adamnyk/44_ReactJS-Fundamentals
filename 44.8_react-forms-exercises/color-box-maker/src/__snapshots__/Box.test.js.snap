// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="Box-box"
      data-testid="colored-box"
      style="background-color: deepskyblue; height: 200px; width: 200px;"
    />
    <button
      class="Box-button"
    >
      x
    </button>
  </div>
</DocumentFragment>
`;
